security/pkg/pki/ra/k8s_ra.go

```diff
@@ -47,9 +47,18 @@ func NewKubernetesRA(raOpts *IstioRAOptions) (*KubernetesRA, error) {
 	return istioRA, nil
 }
 
-func (r *KubernetesRA) kubernetesSign(csrPEM []byte, csrName string, caCertFile string) ([]byte, error) {
+func (r *KubernetesRA) kubernetesSign(csrPEM []byte, csrName string, caCertFile string, certSigner string) ([]byte, error) {
+	certSignerDomain := r.raOpts.CertSignerDomain
+	if r.raOpts.CertSignerDomain != "" {
+		certSignerDomain += "/"
+	}
+	if certSigner == "" {
+		certSigner = certSignerDomain + r.raOpts.CaSigner
+	} else {
+		certSigner = certSignerDomain + certSigner
+	}
```

Comment:
Hmm. I am thinking this may make more sense like:

```suggestion
certSignerDomain := r.raOpts.CertSignerDomain
if certSignerDomain != "" && certSigner != "" {
	certSigner = certSignerDomain+ "/" + certSigner
} else {
	certSigner = r.raOpts.CaSigner
}
```

Basically:
* use r.raOpts.CaSigner directly, not adding domain. This is owned by the control plane admin, so they can configure it however they want
* If CertSignerDomain is not set, do not allow CSR to overwrite the certSigner. This ensures that the control plane has opted into allowing CSR overwrite; otherwise this opens a security hole that anyone can request any signer, which is dangerous if Istiod happens to run with higher RBAC privileges than strictly required.

Reply:
+1 this..Control Plane needs to explicitly buy into the agent selecting its own signer. I would log an error if an agent  wants to use a signer and the control-plane doesn't have a CertSignerDomain set.

Reply:
fixed!  Also logged error when CertSignerDoamin is empty but certSigner is defined.